### PR TITLE
feat: add a receive timeout process for aw_state

### PR DESCRIPTION
**Pull request by v-sugahara7993-esol**
## Description

This PR adds a timeout process when the topic `/in_parking/state` is no longer being received.
If it times out, it is considered an error.

## Related links

* Jira ticket
    * [Ticket (COMPANY INTERNAL LINK)]

[Ticket (COMPANY INTERNAL LINK)]: https://tier4.atlassian.net/browse/T4PB-26456

## Tests performed

The tests described below were conducted.

* Verified that the package can be built without Error/Warning by `colcon`.
* Verified what was described in the Test Case.

<details><summary>Test Case</summary>
<p>

| No |Description|
|----|---|
| 1 |The log detecting timeout should be output after 200ms or more has elapsed since /in_parking/state can no longer be received. |
| 2 |Keep sending errors to the infrastructure when timeouts are detected.|
| 3 |Check that the logs added in this PR are output.|
| 4 |After /in_parking/state is interrupted and a timeout is detected, error transmission shall continue even after /in_parking/state returns.|

</p>
</details>

## Notes for reviewers

None.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by v-sugahara7993-esol** [comment]
The following policy will be applied:

- Implement a timer to monitor the reception status of `/in_parking/state`.
  - When a timeout is detected, the following processing is performed.
    - Update the member variable `aw_state_` to `AW_EMERGENCY`.
    - Stop subscriber of `/in_parking/state`.
    - Output the error log when a timeout is detected.
- The timer is always running, and only while the equipment coordination service is running, it determines whether EMEGENCY should actually be set or not. (This also serves as a countermeasure to prevent misjudgment from initial startup to initial topic reception.)
- The callback group for the monitoring timer is the same as the group for Subscriber and other groups other than the service.
- Add the following processing to `onInParkingStatus()`.
  - Perform the following processing when `AW_EMERGENCY` is received as a message.
  - Stop the subscriber itself in `/in_parking/state`.
- Unify the function used for timeout measurement to `seconds()`.
- Restore the processing of `onTimer()` to the way it was before the change.

File: src/cargo_loading_service.cpp [hunk 1]
```diff
@@ -163,11 +167,33 @@ void CargoLoadingService::onTimer()
...
+    return;
+  }
+
+  auto receive_time_diff = get_clock()->now() - aw_state_last_receive_time_;
+  if (receive_time_diff.seconds() > timeout_time_) {
+    aw_state_ = InParkingStatus::AW_EMERGENCY;
+    sub_inparking_status_ = nullptr;
+    aw_state_timeout_ = true;
+    RCLCPP_ERROR(
+      this->get_logger(), "/in_parking/state timeout detected. aw_state_last_receive_time_ = %lf",
```
Comment (sfukuta) [inline src/cargo_loading_service.cpp]:
```suggestion
      this->get_logger(), "/in_parking/state receive timeout. Last received time (seconds) = %lf",
```

**Review by sfukuta — CHANGES_REQUESTED** [review]
今回追加したログについては、正常に出力できているか確認ずみでしょうか？

**Comment by v-sugahara7993-esol** [comment]
> 今回追加したログについては、正常に出力できているか確認ずみでしょうか？

こちら確認済みです。タイムアウトを検出したタイミングで1回のみログが出力されます。
ログ出力について確認する内容は、本PR文章のTest Case No.3に追加致しました。

**Review by sfukuta — APPROVED** [review]

**Comment by yn-mrse** [comment]
引き継ぎますが、変更権限がないためいったんfeatureへマージします。

**Pull request merged**